File: ProjektyStudentow/2019_Mazepa_Mariusz/src/main/java/decorator/Decorator.java
abstract class Decorator extends LibraryItemPrototype {
  protected LibraryItemPrototype libraryItem;

  public Decorator(LibraryItemPrototype libraryItem) {
    this.libraryItem = libraryItem;
  }

  @Override
  public String toString() {
    return libraryItem.toString();
  }
}

File: old_version/07Dekorator/02Library/Decorator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _02Library {
	abstract class Decorator : Item {
		protected LibraryItem libraryItem;

		protected Decorator(LibraryItem libraryItem) {
			this.libraryItem = libraryItem;
		}

		public override void Display() {
			libraryItem.Display();
		}
	}
}
